### PR TITLE
Add DeepSeek V3.2 decode front scope4 kernel

### PR DESCRIPTION
**Pull request by sjduan**
## Summary

This PR adds a standalone post-topk boundary kernel for DeepSeek V3.2 EXP decode front scope4.

The kernel consumes:
- Rotated `q_proj` (from scope1)
- `topk_idx`, `kv_cache`, and `pe_cache`

And outputs to the cross-node dispatch buffer.

### Key Features

- **5-stage kernel pipeline**:
  1. Load `q_pe` and project `q_nope` into latent space
  2. Run sparse online softmax accumulation in latent space
  3. Project latent context back to V chunks and assemble the row
  4. Cast and stash the finished attention row
  5. Route the finished row into the cross-node dispatch buffer

- **Chunked matmul** to fit platform buffer limits:
  - `Q_LATENT_CHUNK=128` for q_nope_to_latent projection
  - `V_OUT_CHUNK=16` for latent_to_v projection

- **Online softmax** with duplicated 16-row intermediates for a2a3 backend compatibility

- **PyTorch golden reference** for validation

- **Two profile configurations**:
  - `reduced`: batch=16, max_seq=128, num_heads=16 (fast validation)
  - `full`: batch=1, max_seq=4096, num_heads=128 (production scale)

### Usage

```bash
python examples/models/deepseek_v3_2/deepseek_v3_2_decode_front_scope4.py \
    --platform a2a3 \
    --device 0 \
    --profile full \
    --runtime-profiling
```

## Test Plan

- [ ] Run with `--profile reduced` on a2a3sim (fast local validation)
- [x] Run with `--profile full` on a2a3 (production scale validation)
- [x] Verify golden reference matches device output (rtol=1e-3, atol=1e-3)

### CODE REVIEW

File: examples/models/deepseek_v3_2/deepseek_v3_2_decode_front_scope4.py [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+                            )
+                            q_nope_latent_part = pl.matmul(
+                                q_nope_padded,
+                                w_qn_h,
+                                out_dtype=pl.FP32,
+                            )
+                            q_nope_latent_batch = pl.assemble(q_nope_latent_batch, q_nope_latent_part, [0, q0])
+
+                    with pl.at(level=pl.Level.CORE_GROUP):
+                        cache_s0 = b * max_seq_cfg
```
[the remark this answers is not in the document]
Comment (sjduan) [reply examples/models/deepseek_v3_2/deepseek_v3_2_decode_front_scope4.py]: Waiting for `topk_idx` dynamic acquire fix. Current situation:

- the standalone validation harness still fills `topk_idx` with the dense
    window `[0..sparse_k)` instead of the arbitrary device-side `topk_idx`
    consumption that ds32exp Scope 4 intends
    - Observation: read `topk_idx[b, 0]` and `topk_idx[b, kk]` directly.
        - first element read is lowerable: the generated PTO contains a legal 
            `pto.load_scalar %arg2[%arg6 * %c16_index + %c0_index]`.
        - looped reads with the dynamic second index are not lowerable in this 
            shape `compute_ctx_latent.pto` fails with `expected ']'`, the broken 
            point appears when lowering the repeated topk_idx[b, kk] scalar load 
            inside the helper loop.
